From 6b29f598776f8a2ecba09674b4c900ce0fb1fe7f Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Thu, 21 Aug 2014 22:40:56 +1000
Subject: [PATCH 1/3] RingBuffer: Properly wait for readahead thread to start

Inverted test, no wait occurred causing a race condition later on which causes failures in LiveTV, usually when changing channels
---
 mythtv/libs/libmythtv/ringbuffer.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/ringbuffer.cpp b/mythtv/libs/libmythtv/ringbuffer.cpp
index f7651fa..0b855fa 100644
--- a/mythtv/libs/libmythtv/ringbuffer.cpp
+++ b/mythtv/libs/libmythtv/ringbuffer.cpp
@@ -689,7 +689,7 @@ void RingBuffer::Start(void)
 
     MThread::start();
 
-    while (readaheadrunning && !reallyrunning)
+    while (!readaheadrunning && !reallyrunning)
         generalWait.wait(&rwlock);
 
     rwlock.unlock();
-- 
1.7.10.2

